### DISCUSSION TITLE
Ajuda no Firebase

### DISCUSSION THREAD

**Discussion by marcellepaiva** [frontendbr/forum#2412] [Perguntas e Respostas]
Olá devs, recentemente usei o Firebase como banco de dados de um projeto, mas o cloud firestore acabou sendo bloqueado e não sei como resolver esse problema. 

Os emails que tenho recebido do Firebase dizem o seguinte: "Você optou por iniciar o desenvolvimento no Modo de teste, o que deixa seu banco de dados do Cloud Firestore completamente aberto à Internet. Como seu aplicativo ficou vulnerável a ataques, as regras de segurança do Firestore foram configuradas para impedir a permissão de solicitações após os primeiros 30 dias."

Nunca tinha usado esse banco de dados portanto não entendo muito sobre. Se alguém souber como resolver eu agradeço :)

**Comment by heliomarpm** [comment]
> Olá devs, recentemente usei o Firebase como banco de dados de um projeto, mas o cloud firestore acabou sendo bloqueado e não sei como resolver esse problema. 
> 
> Os emails que tenho recebido do Firebase dizem o seguinte: "Você optou por iniciar o desenvolvimento no Modo de teste, o que deixa seu banco de dados do Cloud Firestore completamente aberto à Internet. Como seu aplicativo ficou vulnerável a ataques, as regras de segurança do Firestore foram configuradas para impedir a permissão de solicitações após os primeiros 30 dias."
> 
> Nunca tinha usado esse banco de dados portanto não entendo muito sobre. Se alguém souber como resolver eu agradeço :)
> 

O email fala de uma restrição de acesso que você precisa configurar ao usar o Firestone, nós primeiros meses de desenvolvimento ele até deixa vc usar sem restrições, depois disso vc precisa se conectar para poder ler e escrever na base.

Essas são as regras que funcionam pra mim.

rules_version = '2';
service cloud.firestore {
  match /databases/{database}/documents {
    match /{document=**} {
      // allow read, write: if request.time < timestamp.date(2020, 9, 11);
      allow read, write: if request.auth.uid != null;
    }    
  }
}
